### PR TITLE
Change cursor icon when user's cursor is over canvas image

### PR DESCRIPTION
**Pull request by phil-plencner-hl**
This PR solves this issue: https://github.com/ProjectMirador/mirador/issues/3325

This PR adds viewer handlers for the [canvas-press](https://openseadragon.github.io/docs/OpenSeadragon.Viewer.html#.event:canvas-press) and [canvas-release](https://openseadragon.github.io/docs/OpenSeadragon.Viewer.html#.event:canvas-release) OpenSeadragon events then adds that value to the state. 

The state determines which [CSS cursor icon](https://developer.mozilla.org/en-US/docs/Web/CSS/cursor) is used (`grab` or `grabbing`) in the style attribute of the `<section>` tag.

### CODE REVIEW

File: src/components/OpenSeadragonViewer.js [hunk 1]
```diff
@@ -58,6 +58,14 @@ export class OpenSeadragonViewer extends Component {
 
     this.setState({ viewer });
 
+    viewer.addHandler('canvas-press', () => {
```
Comment (cbeer) [inline src/components/OpenSeadragonViewer.js:61]: would `canvas-drag` and `canvas-drag-end` be more appropriate events?

File: src/components/OpenSeadragonViewer.js [hunk 2]
```diff
@@ -359,6 +367,7 @@ export class OpenSeadragonViewer extends Component {
     return (
       <section
         className={classNames(ns('osd-container'), classes.osdContainer)}
+        style={{ cursor: grabbing ? 'grabbing' : 'grab' }}
```
Comment (cbeer) [inline src/components/OpenSeadragonViewer.js:370]: Could we push the `grab` default up to the osdContainer styles?

File: src/components/OpenSeadragonViewer.js (hunk 1 above)
Comment (phil-plencner-hl) [reply src/components/OpenSeadragonViewer.js:61]: @cbeer: Seems to work the same for me in my browsers. Is that better for touch screen devices? If so, I can commit the change.

**Comment by cbeer** [comment]
I'm seeing something funny: it looks like the grab cursor shows up in the canvas navigation pane too:

Screen Shot 2023-02-22 at 12 49 32

I think we need to turn the cursor back to its default there, because the canvas isn't draggable at that point.

File: src/components/OpenSeadragonViewer.js (hunk 2 above)
Comment (phil-plencner-hl) [reply src/components/OpenSeadragonViewer.js:370]: Done with this commit: https://github.com/ProjectMirador/mirador/pull/3644/commits/b02ff305ba68c466c91f73e294345754b0e26e53

File: src/containers/OpenSeadragonViewer.js [hunk 3]
```diff
@@ -58,6 +58,7 @@ const mapDispatchToProps = {
 const styles = {
   osdContainer: {
     flex: 1,
+    pointer: 'grab',
```
Comment (marlo-longley) [inline src/containers/OpenSeadragonViewer.js:61]: do we want to use the 'cursor' selector?

Comment (phil-plencner-hl) [reply src/containers/OpenSeadragonViewer.js]: Whoops!
https://github.com/ProjectMirador/mirador/pull/3644/commits/4d79018b9434b78c3828248527f143277b986947

File: src/components/OpenSeadragonViewer.js (hunk 2 above)
Comment (phil-plencner-hl) [reply src/components/OpenSeadragonViewer.js:370]: And this commit:
https://github.com/ProjectMirador/mirador/pull/3644/commits/4d79018b9434b78c3828248527f143277b986947

File: src/containers/OpenSeadragonViewer.js (hunk 3 above)
Comment (phil-plencner-hl) [reply src/containers/OpenSeadragonViewer.js]: navigation pane fixed with this commit:
https://github.com/ProjectMirador/mirador/pull/3644/commits/0fc4c3b2198c9ee987a474ae97fb150c2ae89915

File: src/components/OpenSeadragonViewer.js (hunk 1 above)
Comment (phil-plencner-hl) [reply src/components/OpenSeadragonViewer.js]: This commit changes it to use `canvas-drag` and `canvas-drag-end`:
https://github.com/ProjectMirador/mirador/pull/3644/commits/fd2681febf0fb3d6c01148cc2a271b740b4a6aa4

File: src/components/OpenSeadragonViewer.js (hunk 2 above)
Comment (cbeer) [inline src/components/OpenSeadragonViewer.js:370]: Now that the `grab` style is upstream, we can just omit it here:

```suggestion
        style={{ cursor: grabbing ? 'grabbing' : undefined }}
```

Comment (phil-plencner-hl) [reply src/components/OpenSeadragonViewer.js]: Done!
https://github.com/ProjectMirador/mirador/pull/3644/commits/891cc827d949ec7a3ea5d9a7f5559424726a281f

**Pull request merged**